### ISSUE TITLE
KMW Mobile-mode:  Backspace triggering inconsistencies

### ISSUE THREAD

**Issue by jahorton** [keymanapp/keyman#716]
From a colleague:

> I'm testing a keyboard (sil_ipa touch layout) in Google Chrome (version 63.0.3239.73) on an iphone4s with iOS 9.3.5 (using a keyboard served by Keyman Developer 10.0.1055.0, if that matters). Occasionally when I tap the backspace key, I get a very rapid multiple character delete. (Normally if I tap the backspace key, I get a single character delete, or if I tap and hold I get a slow multiple character delete.) I can't reliably reproduce this, but it is disconcerting!

> This is in Chrome, page with heading "Keyboard Test Host". I haven't tried installing the keyboard in Keyman for iOS. So, I'm guessing it is using KMW. It may be that what I thought was a single tap was actually seen as a double tap.

However, another colleague has noticed it as well, including on iOS.

**Comment by jahorton** [comment]
The code being touch-based backspace processing had changed with #585 in an effort to centralize OSK backspace handling (to ensure BKSP-targeted rules were handled uniformly) and allow it to be recorded for generation of automated test sequences, which was merged about a month prior.

Version 10.0.1055 of Developer (an alpha) was released a couple days before this issue was noticed, so the Keyboard Test Host would have been using #585's code.

**Comment by jahorton** [comment]
While testing #818, I saw something highly related to this issue within the iOS app.  After releasing the backspace key after a few presses, the key acted as if I were still holding it down, leading to very rapid character deletions.  It's pretty clear that the app never handled the touch-release event that should have triggered when I released the key, as when I manually touched the key again briefly and released, the deletions stopped.

The question is then why the touch-release event was never handled, with a follow-up question of whether or not it's possible to detect this scenario if touch-release events are unreliable.
